scripts/ai2-internal/run_with_beaker.py

```diff
@@ -56,21 +62,31 @@ def main(param_file: str, args: argparse.Namespace):
                                         universal_newlines=True).strip()
         print(f"  Image created: {docker_image}")
 
-    config_dataset_id = subprocess.check_output(f'beaker dataset create --quiet {param_file}', shell=True, universal_newlines=True).strip()
+    config_dataset_id = subprocess.check_output(f'beaker dataset create --quiet {compiled_params_path}', shell=True, universal_newlines=True).strip()
 
-    allennlp_command = [
-            "python",
-            "-m",
-            "allennlp.run",
-            "train",
-            "/config.json",
-            "-s",
-            "/output",
-            "--file-friendly-logging"
+    # Arguments that differ between preemptible and regular machine execution.
+    if args.preemptible:
+        allennlp_prefix = ["/stage/allennlp/resumable_train.sh", "/output", "/config.json"]
```

Comment:
Currently the base docker image doesn't have this script I think, only the pip Dockerfile. Is that a problem? I think this script supports building your own image - I don't know if you can raise an error or something if `premptible` is passed without specifying a particular image that contains that script or something ...

Reply:
Maybe I'm misunderstanding, but I see it in both `Dockerfile` and `Dockerfile.pip`.
https://github.com/allenai/allennlp/blob/master/Dockerfile#L68
https://github.com/allenai/allennlp/blob/master/Dockerfile.pip#L34

If someone does define a Dockerfile without `resumable_train.sh` and they use `--preemptible`, they'll get an error as soon as they run. This seems reasonable to me. My rationale is that there are many other dependencies that someone could forget to include. We could add checks one by one, but this will be error prone and extra code. Just running the image is simpler and more robust.

Reply:
Oh! completely missed that when I checked the Dockerfile. LGTM.